src/pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -283,6 +283,16 @@ def _check_config_and_set_final_flags(
                 f" Available names are: {', '.join(self._accelerator_types)}."
             )
 
+        # mps accelerator incompatible with ddp-family
+        is_ddp_str = isinstance(strategy, str) and "ddp" in strategy
+        if (
+            accelerator is not None
+            and strategy is not None
```

Comment:
These 2 are not required, no? The `isinstance(None, str)` will shortcut

```suggestion
```

Reply:
Incorporated this.